src/main/java/net/minecraftforge/fml/common/Loader.java

```diff
@@ -601,6 +601,7 @@ public void preinitializeMods()
             FMLLog.warning("There were errors previously. Not beginning mod initialization phase");
             return;
         }
+        PersistentRegistryManager.fireCreateRegistryEvents();
```

Comment:
the NewRegistry event needs to be fired before findObjectHolders for it to resolve custom registries.